mlflow/models/cli.py

```diff
@@ -270,3 +272,51 @@ def build_docker(**kwargs):
     'python_function' flavor.
     """
     python_api.build_docker(**kwargs)
+
+
+@commands.command("update-pip-requirements")
+@cli_args.MODEL_URI
+@click.argument("operation", type=click.Choice(["add", "remove"]))
+@click.argument("requirement_string", type=str)
+def update_pip_requirements(model_uri, operation, requirement_string):
+    """
+    Add or remove requirements from a model's conda.yaml and requirements.txt files.
+    If using a remote tracking server, please make sure to set the MLFLOW_TRACKING_URI
+    environment variable to the URL of the desired server.
+
+    REQUIREMENT_STRING is a comma-separated string of pip requirements,
+    e.g. "pandas==1.0.0, scikit-learn".
+
+    Sample usage:
+
+    .. code::
+
+        # Add requirements using the model's "runs:/" URI
+
+        mlflow models update-pip-requirements -m runs:/<run_id>/<model_path> \\
+            add "pandas==1.0.0, scikit-learn"
+
+        # Remove requirements from a local model
+
+        mlflow models update-pip-requirements -m /path/to/local/model \\
+            remove "torchvision, pydantic"
+
+    Note that model registry URIs (i.e. URIs in the form ``models:/``) are not
+    supported, as artifacts in the model registry are intended to be read-only.
+    Editing requirements is read-only artifact repositories is also not supported.
+
+    If adding requirements, the function will overwrite any existing requirements
+    that overlap, or else append the new requirements to the existing list.
+
+    If removing requirements, the function will ignore any version specifiers,
+    and remove all the specified package names. Any requirements that are not
+    found in the existing files will be ignored.
+    """
+    try:
+        requirements = [Requirement(s.strip().lower()) for s in requirement_string.split(",")]
```

Comment:
`split(",")` doesn't work for the following in the official example:

https://pip.pypa.io/en/stable/reference/requirements-file-format/#example

```
requests [security] >= 2.8.1, == 2.8.* 
```

It's possible that multiple version specifiers exist for a single package.

Reply:
We might need to use a different character as a delimiter (e.g. `|`)

Reply:
or we can make `requirement_string` accept multiple values like:

```
mlflow models update-pip-requirements ... 'foo>2,<3' 'bar>4,<5'
```

Reply:
Makes sense, i'll go with the multiple values approach!